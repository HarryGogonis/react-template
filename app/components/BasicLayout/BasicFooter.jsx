import React from 'react';

import styles from './BasicFooter.styles';

const AuthFooter = () =>
  <div style={styles}>
    <ul style={styles.list}>
      <li style={styles.list.listItem}>
        <a href="http://sharpspring.com/legal/terms" style={styles.list.listItem.link}>Terms of Service</a>
      </li>
      <li style={styles.list.listItem}>
        <a href="http://sharpspring.com/legal/privacy" style={styles.list.listItem.link}>Privacy Policy</a>
      </li>
    </ul>

    <p style={styles.copyright}>Copyright Â© 2012 - 2016</p>
  </div>;

export default AuthFooter;
