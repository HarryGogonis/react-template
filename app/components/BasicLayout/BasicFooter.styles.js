export default {
  padding: '0 10px',
  display: 'flex',
  flexDirection: 'row',
  color: '#fff',
  borderTop: '1px solid rgba(0, 0, 0, 0.1)',
  textTransform: 'uppercase',
  fontSize: '0.75em',
  background: '#3f595f',
  list: {
    margin: '12px 0',
    flex: 1,
    padding: 0,
    listStyle: 'none',
    flexDirection: 'row',
    display: 'flex',
    listItem: {
      marginRight: '10px',
      link: {
        color: '#A6D4DE',
        textDecoration: 'none',
      },
    },
  },
};
