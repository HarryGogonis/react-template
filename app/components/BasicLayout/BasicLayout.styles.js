import { Colors } from 'styleguide';

export default {
  display: 'flex',
  flex: '1',
  height: '100%',
  flexDirection: 'column',
  backgroundColor: Colors.lightGrey,
  backgroundImage: `linear-gradient(to top, ${Colors.lightGrey}, #fff)`,
  content: {
    background: 'rgba(255, 255, 255, 0.5)',
    border: '1px solid rgba(0, 0, 0, 0.1)',
    margin: '0 auto auto',
    position: 'relative',
  },
  logo: {
    height: '100px',
    width: '400px',
    alignSelf: 'center',
    margin: '30px auto 15px',
  },
  copyright: {
    margin: '5px 0',
  },
};
