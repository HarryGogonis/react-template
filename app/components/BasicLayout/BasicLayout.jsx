import React, { PropTypes } from 'react';

import { Images } from 'styleguide';

import BasicFooter from './BasicFooter';
import styles from './BasicLayout.styles';

const propTypes = {
  children: PropTypes.element.isRequired,
};

const BasicLayout = ({ children }) =>
  <div style={styles}>
    <img src={Images.logo} alt="logo" style={styles.logo} />
    <div style={styles.content}>
      {children}
    </div>
    <BasicFooter />
  </div>;

BasicLayout.propTypes = propTypes;

export default BasicLayout;
