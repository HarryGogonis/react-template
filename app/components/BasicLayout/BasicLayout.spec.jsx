import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BasicLayout from './BasicLayout';

jest.mock('styleguide');
jest.mock('./BasicLayout.styles');
jest.mock('./BasicFooter', () => 'BasicFooter');

let wrapper;
const child = <div id="child">Child Content</div>;
const component = (<BasicLayout>
  {child}
</BasicLayout>);

describe('<BasicLayout />', () => {
  beforeEach(() => {
    wrapper = shallow(component);
  });

  it('displays the child node passed', () => {
    expect(wrapper.find('#child').length).toBe(1);
  });

  it('has a img (for logo)', () => {
    expect(wrapper.find('img').length).toBe(1);
  });

  it('has a BasicFooter component', () => {
    expect(wrapper.find('BasicFooter').length).toBe(1);
  });

  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
