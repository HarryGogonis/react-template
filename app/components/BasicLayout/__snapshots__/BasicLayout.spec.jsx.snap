// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicLayout /> should match the snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#D1D6D8",
      "backgroundImage": "linear-gradient(to top, #D1D6D8, #fff)",
      "content": Object {
        "background": "rgba(255, 255, 255, 0.5)",
        "border": "1px solid rgba(0, 0, 0, 0.1)",
        "margin": "0 auto auto",
        "position": "relative",
      },
      "copyright": Object {
        "margin": "5px 0",
      },
      "display": "flex",
      "flex": "1",
      "flexDirection": "column",
      "height": "100%",
      "logo": Object {
        "alignSelf": "center",
        "height": "100px",
        "margin": "30px auto 15px",
        "width": "400px",
      },
    }
  }
>
  <img
    alt="logo"
    src="test-file-stub"
    style={
      Object {
        "alignSelf": "center",
        "height": "100px",
        "margin": "30px auto 15px",
        "width": "400px",
      }
    }
  />
  <div
    style={
      Object {
        "background": "rgba(255, 255, 255, 0.5)",
        "border": "1px solid rgba(0, 0, 0, 0.1)",
        "margin": "0 auto auto",
        "position": "relative",
      }
    }
  >
    <div
      id="child"
    >
      Child Content
    </div>
  </div>
  <BasicFooter />
</div>
`;
